src/core/lib/iomgr/ev_epollex_linux.cc

```diff
@@ -677,6 +679,11 @@ static grpc_error* pollset_kick_all(grpc_pollset* pollset) {
 static void pollset_init(grpc_pollset* pollset, gpr_mu** mu) {
   gpr_mu_init(&pollset->mu);
   pollset->active_pollable = POLLABLE_REF(g_empty_pollable, "pollset");
+  pollset->kicked_without_poller = false;
+  pollset->shutdown_closure = nullptr;
+  pollset->already_shutdown = false;
```

Comment:
In `udp_server_test`,  the pollset object is reused several times (grpc_pollset_init after it has been shutdown previously), which would break the test because already_shutdown would stay `true` here and the test would hang.
To fix, initializing all pollset fields explicitly (other pollers also initialize their fields explicitly, so this is also good for consistency).

Reply:
`shutdown_closure` was being to set to `nullptr` once the shutdown is finished, `already_shutdown` was something that was added in this PR. That leaves `kicked_without_poller` which is also set to `false` whenever it is observed to be true.

So I am not exactly sure where the error is happening.

Reply:
Oh..I think I understand. 
Since `shutdown_closure == nullptr` can mean that the pollset is new or already shutdown , you added `already_shutdown` to disambiguate that. Makes sense.

Resetting them to their default states is not strictly necessary but I am ok with doing this. 

Ignore my previous comment. Thanks.